BlobTierAnalysisTool/BlobTierAnalysisTool.NetCore/Helpers/BlobStorageHelper.cs

```
@@ -136,71 +110,64 @@ public static async Task<IEnumerable<string>> ListContainers()
+                                }
+                                break;
+                            case "Cool":
+                                var coolAccessTierStats = containerStats.BlobsStatistics[AccessTier.Cool];
+                                coolAccessTierStats.Count += 1;
+                                coolAccessTierStats.Size += blobSize;
+                                if (doesBlobMatchFilterCriteria)
+                                {
+                                    var matchingCoolAccessTierStats = containerStats.MatchingBlobsStatistics[AccessTier.Cool];
+                                    matchingCoolAccessTierStats.Count += 1;
+                                    matchingCoolAccessTierStats.Size += blobSize;
+                                    matchingCoolAccessTierStats.BlobNames.Add(blob.Name);
+                                }
+                                break;
+                            case "Archive":
+                                var archiveAccessTierStats = containerStats.BlobsStatistics[AccessTier.Archive];
+                                archiveAccessTierStats.Count += 1;
+                                archiveAccessTierStats.Size += blobSize;
+                                if (doesBlobMatchFilterCriteria)
+                                {
+                                    var matchingArchiveAccessTierStats = containerStats.MatchingBlobsStatistics[AccessTier.Archive];
+                                    matchingArchiveAccessTierStats.Count += 1;
+                                    matchingArchiveAccessTierStats.Size += blobSize;
+                                    matchingArchiveAccessTierStats.BlobNames.Add(blob.Name);
+                                }
+                                break;
                         }
                     }
                 }
-                while (token != null);
             }
-            catch (Exception exception)
+            catch (Exception)
```

Are the statistics that are returned accurate and valid in the face of an exception?  It definitely doesn't seem that way.  I'm not sure that swallowing the exception here is the example that we want to set.

---

Agreed, I don't see why we would swallow this exception and return potentially partially updated data.

---

"containerStats" doesn't change in try catch, So the exception will not affect its value.

---

We are mutating the dictionary values in `containerStats`:
```c#
var matchingArchiveAccessTierStats = containerStats.MatchingBlobsStatistics[AccessTier.Archive];
matchingArchiveAccessTierStats.Count += 1;
matchingArchiveAccessTierStats.Size += blobSize;
matchingArchiveAccessTierStats.BlobNames.Add(blob.Name);
```